proposals/GlobalUsingDirective.md

```diff
@@ -2,13 +2,194 @@
+
+# Extension method invocations
+https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#extension-method-invocations
+
+Changes are made to the algorithm to find the best *type_name* `C` as follows.
+This is the relevant bullet point with proposed additions (which are **in bold**):
+*  Starting with the closest enclosing namespace declaration, continuing with each enclosing namespace declaration, and ending with the containing compilation unit, successive attempts are made to find a candidate set of extension methods:
+   * If the given namespace or compilation unit directly contains non-generic type declarations `Ci` with eligible extension methods `Mj`, then the set of those extension methods is the candidate set.
+   * If types `Ci` imported by *using_static_declarations* and directly declared in namespaces imported by *using_namespace_directive*s in the given namespace or compilation unit directly contain eligible extension methods `Mj`, then the set of those extension methods is the candidate set.
+   * **If containing compilation unit is reached and if types `Ci` imported by *global_using_static_declarations* and directly declared in namespaces imported by *global_using_namespace_directive*s in the program directly contain eligible extension methods `Mj`, then the set of those extension methods is the candidate set.**
+
+# Compilation units
+https://github.com/dotnet/csharplang/blob/master/spec/namespaces.md#compilation-units
+
+A *compilation_unit* defines the overall structure of a source file. A compilation unit consists of **zero or more *global_using_directive*s followed by** zero or more *using_directive*s followed by zero or more *global_attributes* followed by zero or more *namespace_member_declaration*s.
+
+```antlr
+compilation_unit
+    : extern_alias_directive* global_using_directive* using_directive* global_attributes? namespace_member_declaration*
+    ;
+```
+
+A C# program consists of one or more compilation units, each contained in a separate source file. When a C# program is compiled, all of the compilation units are processed together. Thus, compilation units can depend on each other, possibly in a circular fashion.
+
+The *global_using_directive*s of a compilation unit affect the *using_directive*s, *global_attributes* and *namespace_member_declaration*s of all compilation units in the program.
+
+# Extern aliases
+https://github.com/dotnet/csharplang/blob/master/spec/namespaces.md#extern-aliases
+
+The scope of an *extern_alias_directive* extends over the ***global_using_directive*s,** *using_directive*s, *global_attributes* and *namespace_member_declaration*s of its immediately containing compilation unit or namespace body.
```

Comment:
Really not sure I like this behavior. I'd rather see us introduce a global extern alias that can apply to all compilation units, rather than make extern aliases apply to the global aliases in their file.